### ISSUE TITLE
Results for clickbait dataset.

### ISSUE THREAD

**Issue by ashim95** [lethaiq/shield-defend-adversarial-texts#2]
Hi @lethaiq ,

Thank you very much for releasing your code. 

I made one small change to your code: as is common with text classification tasks, I used the `max sentence length = 128` which was previously set to 32. I ran your code with all other default settings on the `clickbait` dataset on `TextFooler` attack and get very different results. For the base model with BERT, I get the following results:
```
Origin accuracy: 0.9875
Adversarial accuracy: 0.35125
Attack Success Rate: 0.28544303797468357
```

For the `shield` model, I get the following numbers:
```
Origin accuracy: 0.924375                                                                                                                                     
Adversarial accuracy: 0.5396875                                                                                                                               
Attack Success Rate: 0.3262339418526031
```
I know you didn't report results for `sst2` from `glue` but I ran the experiments for `sst2` dataset since it is very similar task, and get the following results:
For `base` model with BERT:
```
Origin accuracy: 0.9231651376146789                                                                                                                           
Adversarial accuracy: 0.23738532110091742                                                                                                                     
Attack Success Rate: 0.6832298136645962
```
For the `shield` model:
```
Origin accuracy: 0.8211009174311926                                                                                                                           
Adversarial accuracy: 0.25229357798165136                                                                                                                     
Attack Success Rate: 0.7360335195530726   
```
Did you observe any such results during your experiments? Is it possible to report the exact commands and hyperparameters that you ran to report the results in the paper? 

Finally, shouldn't the `Adversarial accuracy` be `100 - Attack Success Rate`?

Thank you,

**Comment by lethaiq** [comment]
@ashim95 ,

I will have a look at the first part. As for the attack success rate, please look at the `cal_true_success_rate()` under `utils.py` file. Here we only consider an attack is success of the original clean example was correctly predicted. This is to remove bias and ensure faire comparisons as there might be differences in the original prediction accuracy across predictive models.

**Comment by lethaiq** [comment]
@ashim95 
I make some updates to add random seed. I also added new results for clickbait in the main page. Please check and compare with your code. I also edit the max_len to 128 as you tested. I think you only need to make sure the NAS parameters converge to near one-hot vector during training (as noted in the readme file)

**Comment by ashim95** [comment]
Thank you @lethaiq for your response. 

I did notice the NAS parameters being close to one-hot during training. I will evaluate again with different `tau` values and let you know.

**Comment by lethaiq** [comment]
Resolved per email exchange. Please feel free to open if you have more questions @ashim95 !
